### PR TITLE
Fix rendering glitches with EnableDepthBuffer disabled.

### LINKED ISSUE #19621: Images slightly outside of the viewport are cut off
Introduced by ef1aee5e95a79cebc79f120e80fff2c69790e089 the renderer starts to cut of sprites that are not completely inside the viewport.

ra

### PR DESCRIPTION
**Pull request by pchote**
Fixes #19621.

### CODE REVIEW

File: OpenRA.Game/Graphics/SpriteRenderer.cs [hunk 1]
```diff
@@ -227,11 +227,14 @@ public void SetViewportParams(Size sheetSize, int downscale, float depthMargin,
...
+			// * If depthMargin == 0 (which indicates per-pixel depth testing is disabled) sprites that
+			//   extend beyond the top of bottom edges of the screen may be pushed outside [-1, 1] and
+			//   culled by the GPU. We avoid this by forcing everything into the z = 0 plane.
+			var depth = depthMargin != 0f ? 2f / (downscale * (sheetSize.Height + depthMargin)) : 0;
 			shader.SetVec("DepthTextureScale", 128 * depth);
-			shader.SetVec("Scroll", scroll.X, scroll.Y, scroll.Y);
+			shader.SetVec("Scroll", scroll.X, scroll.Y, depthMargin != 0f ? scroll.Y : 0);
 			shader.SetVec("r1", width, height, -depth);
-			shader.SetVec("r2", -1, -1, 1);
+			shader.SetVec("r2", -1, -1, depthMargin != 0f ? 1 : 0);
```
Comment (Mailaender) [inline OpenRA.Game/Graphics/SpriteRenderer.cs:237]: Cache `bool depthBufferEnabled = depthMargin != 0f` for performance?

Comment (pchote) [reply OpenRA.Game/Graphics/SpriteRenderer.cs:237]: performance impact would be negligible, but I guess I could do this if there were a consensus that it helped readability.

**Review by Mailaender — APPROVED** [review]
Fix confirmed. Thanks.

File: OpenRA.Game/Graphics/SpriteRenderer.cs (hunk 1 above)
Comment (penev92) [reply OpenRA.Game/Graphics/SpriteRenderer.cs:237]: Remember kids, always prefix your Boolean fields/properties/methods/variables with "is"/"has"/"should”/"can" 😉 
(I have no preference on the readability question here, but you *are* doing the check three times 🤷‍♂️ )

**Comment by dnqbob** [comment]
Fix confirmed. Thanks for the attention here!

fixed
bf

previous

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/b812658d70a0198ba3f336a788a737caca83fbd0...acb358b636eb6615ddbf2ee22224e40d6dd2b680)

**Pull request merged**